examples/kprobepin/bpf/kprobe_pin_example.c

```diff
@@ -0,0 +1,26 @@
+#include "common.h"
+#include "bpf_helpers.h"
+
+char __license[] SEC("license") = "Dual MIT/GPL";
+
+struct bpf_map_def SEC("maps") kprobe_map = {
```

Comment:
You can simplify this by using a BTF style map definition with `__uint(pinning, LIBBPF_PIN_BY_NAME)`. Then specify [MapOptions.PinPath](https://pkg.go.dev/github.com/cilium/ebpf#MapOptions) and the library will handle things for you.